modules/volume/src/algorithm/volumemap.cpp

```diff
@@ -0,0 +1,107 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2022 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <inviwo/volume/algorithm/volumemap.h>
+
+#include <algorithm>
+#include <unordered_map>
+#include <unordered_set>
+#include <inviwo/core/util/exception.h>
+
+namespace inviwo {
+
+void remap(std::shared_ptr<Volume>& volume, std::vector<int> src, std::vector<int> dst,
+           int missingValue, bool useMissingValue) {
+
+    if (src.size() == 0 || src.size() != dst.size()) {
+        throw Exception(IVW_CONTEXT_CUSTOM("Remap"), "Invalid dataframe size (src = {}, dst = {})",
+                        src.size(), dst.size());
+    }
+
+    // Create sorted copy of src and check if it contains duplicates
+    std::unordered_set<int> set(src.begin(), src.end());
+    if (src.size() != set.size()) {
+        throw Exception(IVW_CONTEXT_CUSTOM("Remap"),
```

Comment:
An option could be to simply remove duplicates.

Reply:
Then it would be unclear what happens, that is which index is reused. This means that there need to be at least a log warning which implies the need for logging (and showing the log). An exception is much clearer.